### PR TITLE
VCVT{T,}PD2DQ fixes and optimization

### PR DESCRIPTION
**Pull request by Sonicadvance1**
A few things here.

- Fixes a bug with VCVT{T,}PD2DQ where it was failing to zero the upper 128-bit lane in AVX128
  - Updating the unittests to ensure it happens
- Adds a new IR path to take advantage of new SVE instructions which can directly F64->I32
- Fixes a /minor/ precision issue reported in #2995 due to truncating F64->F32 before rounding which could have resulting in rounding the wrong direction.

This doesn't change behaviour around overflow problems with conversion. If we want to handle that then we need to enable FP exceptions and capture the overflow and continue running. Which most ARM hardware doesn't support those exceptions anyway. x86 always returns its "indefinite integer" value of 0x8000_0000, where ARM will return the saturated minimum or maximum int32_t depending on sign.

### CODE REVIEW

**Comment by JosJuice** [comment]
I think you still have a rounding problem in the non-SVE path. Using FRINTI before FCVTN does ensure that the value sent to FCVTN is an integer, but there's a possibility that it's an integer that can't be represented exactly in f32 but can be represented in i32. You have to avoid intermediate conversion to f32 in order to solve this, and as far as I know, the best way to do that in ASIMD is to use FCVTZS (scalar, integer) in place of FCVTN (double to single-precision) + FCVTZS (vector, integer).

**Comment by JosJuice** [comment]
I guess there's also the option of converting f64->i64 and then manually narrowing it to i32. This could be convenient if you want to emulate overflow accurately, but manually narrowing would take a few extra instructions.

**Comment by Sonicadvance1** [comment]
> FCVTZS

Indeed, I have thought about reducing down to using the scalar operations but I'm not quite ready to take that leap without more testing. This PR gets us setup to take that plunge if need be

**Comment by Sonicadvance1** [comment]
For future excavation the diff for implementing that code path is as follows:
```diff
diff --git a/FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp b/FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp
index 6156e07b9..f08bfb145 100644
--- a/FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp
+++ b/FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp
@@ -505,11 +505,14 @@ DEF_OP(Vector_F64ToI32) {
     case FEXCore::IR::Round_Host.Val: frinti(ARMEmitter::SubRegSize::i64Bit, Dst.Q(), Vector.Q()); break;
     }

-    // Now narrow from f64 to f32.
-    fcvtn(ARMEmitter::SubRegSize::i32Bit, Dst.Q(), Dst.Q());
-
-    ///< Convert the two F32 integrals to real integers.
-    fcvtzs(ARMEmitter::SubRegSize::i32Bit, Dst.D(), Dst.D());
+    ///< Narrow from F64->I32
+    dup(ARMEmitter::SubRegSize::i64Bit, VTMP1.Q(), Dst.Q(), 1);
+    fcvtzs(ARMEmitter::Size::i32Bit, TMP1.W(), Dst.D());
+    fcvtzs(ARMEmitter::Size::i32Bit, TMP2.W(), VTMP1.D());
+
+    ///< Insert back in to vector register.
+    fmov(ARMEmitter::Size::i32Bit, Dst.S(), TMP1.W());
+    ins(ARMEmitter::SubRegSize::i32Bit, Dst, 1, TMP2.W());
   }
 }
 ```

File: FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp [hunk 1]
```diff
@@ -449,5 +450,68 @@ DEF_OP(Vector_FToI) {
...
+  const auto Dst = GetVReg(Node);
+  const auto Vector = GetVReg(Op->Vector.ID());
+  if (HostSupportsSVE128 || HostSupportsSVE256) {
+    const auto Mask = Is256Bit ? PRED_TMP_32B.Merging() : PRED_TMP_16B.Merging();
+    // First step is to round the f64 values to integrals (frint*)
+    // Then convert to integers using fcvtzs.
+    switch (Round) {
+    case IR::Round_Nearest.Val:
+      frintn(ARMEmitter::SubRegSize::i64Bit, Dst.Z(), Mask, Vector.Z());
+      fcvtzs(Dst.Z(), ARMEmitter::SubRegSize::i32Bit, Mask, Dst.Z(), ARMEmitter::SubRegSize::i64Bit);
```
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp:469]: Pop the fcvtzs out of the switch for deduplication across all the cases

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/JIT/Arm64/ConversionOps.cpp]: Popped

**Review by alyssarosenzweig — APPROVED** [review]

**Comment by lioncash** [comment]
Perhaps document the rounding problem Jos brought up in the ASIMD path so it isn't totally forgotten about. Otherwise this seems fine as an improvement

**Comment by Sonicadvance1** [comment]
> Perhaps document the rounding problem Jos brought up in the ASIMD path so it isn't totally forgotten about. Otherwise this seems fine as an improvement

Added a comment

**Review by lioncash — APPROVED** [review]

**Pull request merged**